### ISSUE TITLE
Form submit action not redirected to default route POST action after upgrade to .NET 6 preview 7

### ISSUE THREAD

**Issue by odk0706** [dotnet/aspnetcore#35384]
### Describe the bug
Form POST action not mapping to controller/action after upgrading .NET 5 to .NET 6 preview 7.

We upgraded our .NET 5 MVC project to 6 preview 7. After the upgrade the login page does show as usual, but the login action does not reach the POST action on the controller. 

I have no idea what could be the reason for this issue. 

### To Reproduce
Startup cs:
We have multiple areas but the login method is located in the last route defined ( the non-area/root controllers). Maybe this mapping has something to do with the issue (please note that the mapping did work correctly in .NET 5). Some text is replaced with *.
```csharp
app.UseEndpoints(endpoints =>
			{

				// Area Routes 
				endpoints.MapAreaControllerRoute(
					"***_dyn_api_route",
					"***DynApi",
			"***Api/{controller=Dynamic}/{action=HandleRequest}/{entityName:alpha:required}/{actionName:alpha:required}");
				endpoints.MapAreaControllerRoute(
					"***_api_route",
					"***Api",
					"***Api/{controller}/{action}/{id?}");
				endpoints.MapAreaControllerRoute(
					"***_app_route",
					"***App",
					"***App/{controller}/{action}/{id?}");
				endpoints.MapAreaControllerRoute(
					"***_sso_route",
					"***Sso",
					"***Sso/{controller=Users}/{action=Index}/{id?}");
				endpoints.MapAreaControllerRoute(
				   "***_***_route",
				   "******",
				   "******/{controller=Home}/{action=Index}/{id?}");
				endpoints.MapAreaControllerRoute(
					"***_payment_route",
					"***",
					"***/{controller=Payment}/{action=Index}/{id?}");
				endpoints.MapAreaControllerRoute(
					"***_***_route",
					"******",
					"******/{controller=Home}/{action=Index}/{id?}");
				endpoints.MapAreaControllerRoute(
					"***_KnowledgeBase_route",
					"***KnowledgeBase",
					"***Knowledgebase/{controller=KnowledgeBase}/{action=Index}/{id?}");

				// The root controllers route (that contains the Login controller)
				endpoints.MapControllerRoute(
					name: "default",
					pattern: "{controller=Login}/{action=Index}/{id?}");
					


				endpoints.MapRazorPages();
			});	
```

Index.cshtml:
````html 
<form asp-action="Index" asp-controller="Login" asp-antiforgery="true">
````

Rendered result:
The rendered HTML is the same in .NET 5 and 6 so the tag helpers did not change.
````html 
<form action="/" method="post">
````

Controller post method:
Not changed after the upgrade. The action is located in the HomeController. Please note that the GET method is working correctly.
````csharp 
[ValidateAntiForgeryToken]
[AllowAnonymous] 
[HttpPost] 
public async Task<IActionResult> Index(SignInModel model){
}
````

**Comment by odk0706** [comment]
![image](https://user-images.githubusercontent.com/4162876/129576721-da7c5a34-2a11-49c0-9c4f-f23b5714dd08.png)

If we change the form action to the path explicitly it works. So it looks like the default route action does not work for POST requests. 

Is there any reason for this change, and if so shouldn't the tag helper render the full path to the action? 

If I need to provide any more information please let me know. Thank you in advance!

**Comment by davidfowl** [comment]
If you can provide a something thing isolated and runnable that would be great. Trying to debug what's going on without most of the application configuration might be tricky.

**Comment by SteveSandersonMS** [comment]
Thanks for the report. Are you able to supply a minimal repro project, ideally as a GitHub issue? We will then investigate.

**Comment by odk0706** [comment]
I tried to isolate the code in a separate (empty .NET 6 MVC) project. Initially, the results were the same, but after deleting the cookies (for localhost:5011) the error disappeared in both projects... 
Unfortunately, I did not save the cookies that were deleted...

Switching the project between .NET 5 and 6 also does not trigger the error. So for now everything is working again.

When we run into this problem again I will be happy to share some more detailed information with you!
Thanks for the time and effort you put into the project 👍
